### PR TITLE
Added planners.

### PR DESCRIPTION
**Pull request by CPWstatic**

### CODE REVIEW

File: src/planner/Planner.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+public:
+    virtual ~Planner() = default;
+
+    static StatusOr<SubPlan> toPlan(Validator* validator);
+
+    virtual bool match(Validator* validator) = 0;
+
+    virtual StatusOr<SubPlan> transform(Validator* validator) = 0;
+
+    auto& plannersMap() {
```
[the remark this answers is not in the document]
Comment (CPWstatic) [reply src/planner/Planner.h:33]: `Status::Error()` returns an instance of `Status`, it seems different with this scenario.

File: src/validator/SequentialValidator.h [hunk 2]
```diff
@@ -8,12 +8,17 @@
...
 #include "common/base/Base.h"
-#include "validator/Validator.h"
 #include "parser/SequentialSentences.h"
+#include "validator/Validator.h"
 
 namespace nebula {
 namespace graph {
 
+struct SequentialAstContext final : AstContext {
+    QueryContext*                               qctx;
```
Comment (yixinglu) [inline src/validator/SequentialValidator.h:18]: Maybe this qctx could be pulled up into AstContext.

File: src/planner/Planner.cpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ */
+
+#include "planner/Planner.h"
+
+#include "validator/Validator.h"
+
+namespace nebula {
+namespace graph {
+
+std::unordered_map<Sentence::Kind, std::vector<Planner*>> Planner::plannersMap_;
```
Comment (Shylock-Hg) [inline src/planner/Planner.cpp:14]: Is it better to manage the Planners by tree or DAG?

**Review by yixinglu — APPROVED** [review]

File: src/planner/Planner.cpp (hunk 3 above)
Comment (CPWstatic) [reply src/planner/Planner.cpp]: Emm, the planners don't have that kind of relation.

**Review by yixinglu — COMMENTED** [review]
LGTM

**Review by dutor — APPROVED** [review]
LGTM.

**Review by yixinglu — APPROVED** [review]

**Pull request merged**